indexer/database/db.go

```diff
@@ -23,36 +26,41 @@ type DB struct {
 }
 
 func NewDB(dbConfig config.DBConfig) (*DB, error) {
-	dsn := fmt.Sprintf("host=%s port=%d dbname=%s sslmode=disable", dbConfig.Host, dbConfig.Port, dbConfig.Name)
-	if dbConfig.User != "" {
-		dsn += fmt.Sprintf(" user=%s", dbConfig.User)
-	}
-	if dbConfig.Password != "" {
-		dsn += fmt.Sprintf(" password=%s", dbConfig.Password)
-	}
-	gorm, err := gorm.Open(postgres.Open(dsn), &gorm.Config{
-		// The indexer will explicitly manage the transaction
-		// flow processing blocks
-		SkipDefaultTransaction: true,
+	var db *DB
+
+	retryStrategy := &retry.ExponentialStrategy{Min: 1000, Max: 20_000, MaxJitter: 250}
+
+	_, err := retry.Do[interface{}](context.Background(), 10, retryStrategy, func() (interface{}, error) {
+		dsn := fmt.Sprintf("host=%s port=%d dbname=%s sslmode=disable", dbConfig.Host, dbConfig.Port, dbConfig.Name)
+		if dbConfig.User != "" {
+			dsn += fmt.Sprintf(" user=%s", dbConfig.User)
+		}
+		if dbConfig.Password != "" {
+			dsn += fmt.Sprintf(" password=%s", dbConfig.Password)
+		}
+		gorm, err := gorm.Open(postgres.Open(dsn), &gorm.Config{
+			SkipDefaultTransaction: true,
+			Logger:                 logger.Default.LogMode(logger.Silent),
+		})
 
-		// We may choose to create an adapter such that the
-		// logger emits to the geth logger when on DEBUG mode
-		Logger: logger.Default.LogMode(logger.Silent),
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to connect to database")
+		}
+
+		db = &DB{
```

Comment:
nit: change `var db *DB` to `var gorm *gorm.DB` so that  we can do the `DB` construction at the end of the function on success